actionview/test/template/template_test.rb

```diff
@@ -149,6 +149,61 @@ def test_encoding_can_be_specified_with_magic_comment
     assert_equal "\nhello \u{fc}mlat", render
   end
 
+  def test_locals_can_be_disabled
+    error = assert_raises(ActionView::Template::Error) do
+      @template = new_template("<%# locals: () -%>")
+      render(foo: "bar")
+    end
+
+    assert_match(/does not accept locals/, error.message)
+  end
+
+  def test_locals_can_not_be_specified_with_positional_arguments
+    error = assert_raises(ActionView::Template::Error) do
+      @template = new_template("<%# locals: (foo) -%>")
+      render(foo: "bar")
+    end
+
+    assert_match(/`foo` set as non-keyword argument/, error.message)
+  end
+
+  def test_locals_can_not_be_specified_with_splat_arguments
+    error = assert_raises(ActionView::Template::Error) do
+      @template = new_template("<%# locals: (bar:, baz: "", **splat) -%>")
+      render(foo: "bar")
+    end
+
+    assert_match(/`splat` set as non-keyword argument/, error.message)
+  end
+
+  def test_locals_can_be_specified
+    @template = new_template("<%# locals: (message:) -%>\n<%= message %>")
+    assert_equal "Hello", render(message: "Hello")
+  end
+
+  def test_default_locals_can_be_specified
+    @template = new_template("<%# locals: (message: 'Hello') -%>\n<%= message %>")
+    assert_equal "Hello", render
+  end
+
+  def test_required_locals_can_be_specified
+    error = assert_raises(ActionView::Template::Error) do
+      @template = new_template("<%# locals: (message:) -%>")
+      render
+    end
+
+    assert_match(/missing keyword: :message/, error.message)
```

Comment:
Hum, should we wrap these exception so that is says "missing local" instead of "missing keyword"? Maybe it's too much?

Reply:
🤔 it doesn't bother me too much, but I see your point. How would you suggest building this? Rescuing `ArgumentError` in `Base#_run`?

Reply:
Yup